kernel/poll.c

```diff
@@ -292,7 +292,8 @@ static int _signal_poll_event(struct k_poll_event *event, u32_t state,
 
 	_unpend_thread(thread);
 	_abort_thread_timeout(thread);
-	_set_thread_return_value(thread, 0);
+	_set_thread_return_value(thread,
+				 state == K_POLL_STATE_NOT_READY ? -EINVAL : 0);
```

Comment:
Is EINVAL a good error here? It's meaning is "invalid *input* value" or "invalid value", there's nothing invalid in K_POLL_STATE_NOT_READY, it's just a special value to act upon. Previously, EINTR was mentioned, or we definitely use EAGAIN in somewhat similar contexts.

Reply:
Would -ECANCELED be more relevant?

Reply:
Changed that to -EINTR, for whatever reason I thought that was not available on zephyr.